cluster-creation-cli/README.md

```diff
@@ -0,0 +1,163 @@
+
+Ex. 5: Write the result to a file
+
+    kubectl gs create cluster --provider capa --config myconfig.yaml --output manifest.yaml
+
+## Configuration layering
+
+Configuration can be passed to the command in several ways:
+
+1. Any number of YAML files specified via the `--config` flag.
+2. YAML passed to standard input (STDIN).
+3. Any number of single values defined via the `--set` flag.
+
+The command will create one ConfigMap per source, and one [App](https://docs.giantswarm.io/vintage/use-the-api/management-api/crd/apps.application.giantswarm.io/) resource for the cluster-app. In the App resource, the ConfigMaps will be referenced under the `.spec.extraConfigs` property, which is an array. This way, the number of ConfigMaps is not limited.
+
+The individual ConfigMaps' priority will be configured explicitly in the following order, where later sources overwrite earlier sources:
+
+1. Each file specified via the `--config` flag, in the order they are specified.
+2. Standard input (STDIN).
+3. Each single value specified via the `--set` flag, in the order they are specified.
+
+The ConfigMap representing STDIN always comes last and takes highest priority. This allows to override configuration form one file with a another file specified later in the chain, or with a value passed via STDIN.
+
+In other words, we leverage the App platform's configuration layering principle to provide a flexible templating mechanism, where the underlying layer is always provided by the app's default values.
+As a usage example, this layering could be used to create a cluster using a template configuration file, and overriding some specific values via STDIN:
+
+    kubectl gs create cluster --provider capa --config template.yaml <<EOF
+    global:
+      metadata:
+        name: test02
+    EOF
+
+Alternative syntax for above example, using the `--set` flag:
+
+    kubectl gs create cluster --provider capa --config template.yaml \
+      --set global.metadata.name=test02
+
+This enables the target scenario as described earlier, by re-using most configuration values from one file, and providing the required difference (here: the cluster name) via standard input. The same could of course be achieved by passing a second YAML file via another `--config` flag instead.
+
+In input validation, each source is evaluated against the app's values schema independently. We avoid replicating any values merging logic (as known from helm or the app platform) into the kubectl-gs CLI code base.
+
+## Progress display during creation
+
+If the user chooses to create the cluster immediately (by not specifying an output destination or format), we want to provide detailed progress on the creation. The user should be encouraged to quit watching the progress at any time, without any effect on the provisioning progress.
+
+We may even introduce another subcommand to continue watching the creation progress at any time.
+
```

Comment:
This RFC seems to be missing anything about how a cluster would be upgraded/modified after creation 
(can I re-run the create command and it'll overwrite? Will it block me because a cluster with the same name already exists? If it doesn't block, how can I avoid accidentally overwriting someone else's cluster named `test`?)

I think this RFC needs to discuss how / if / where this command would fit into a gitops workflow. We're pretty actively encouraging our customers to use gitops and this approach seems to go against that advice.